### ISSUE TITLE
Core: Replace Container.__setitem__ with public Container.replace() method

### ISSUE THREAD

**Issue by trevorbaca** [Abjad/abjad#463]
Too many `Container.__setitem__()` behaviors are implicit. Why? Because putting music into a (potentially very complex) container in score is more involved then putting items in a built-in list.

We should instead replace `Container.__setitem__()` with a public method. This will then enable the method to accept a collection of keywords. Each keyword can be documented. And then we can (finally) cleanly document the different set-item behaviors: are spanners fractured? should music be replaced or kept? what about indicators? And so on.

Basically the problem is that `__setitem__()` can't take keywords.

**Comment by trevorbaca** [comment]
In fact we have such a public method: ``mutate().replace()``.

We do not need to implement a ``Container.replace()`` method. We just need to test to make sure that ``mutate().replace()`` does everything that ``Container.__setitem__()`` currently does. Once we're sure that ``mutate().replace()`` does everything that ``Container.__setitem__()`` does then we can remove ``Container.__setitem__()``.

**Comment by trevorbaca** [comment]
See #454 too.

**Comment by trevorbaca** [comment]
Still valid in June 2017.

**Comment by trevorbaca** [comment]
Duplicate of #535.
